core/chains/evm/client/client.go

```diff
@@ -197,6 +202,107 @@ func (client *client) HeaderByHash(ctx context.Context, h common.Hash) (*types.H
 	return client.pool.HeaderByHash(ctx, h)
 }
 
+func (client *client) SendTransactionAndReturnErrorType(ctx context.Context, tx *types.Transaction, fromAddress common.Address) (txmtypes.TxmErrorType, error) {
+	err := client.pool.SendTransaction(ctx, tx)
+	sendError := NewSendError(err)
+	if sendError.Fatal() {
+		client.logger.Criticalw("Fatal error sending transaction", "err", sendError, "etx", tx)
+		// Attempt is thrown away in this case; we don't need it since it never got accepted by a node
+		return txmtypes.Fatal, err
+	}
+	if sendError.IsNonceTooLowError() || sendError.IsTransactionAlreadyMined() {
+		// Nonce too low indicated that a transaction at this nonce was confirmed already.
+		// Mark confirmed_missing_receipt and wait for the next cycle to try to get a receipt
+		return txmtypes.SuccessfulMissingReceipt, err
+	}
+	if sendError.IsReplacementUnderpriced() {
+		client.logger.Errorw(fmt.Sprintf("Replacement transaction underpriced for eth_tx %x. "+
+			"Eth node returned error: '%s'. "+
+			"Please note that using your node's private keys outside of the chainlink node is NOT SUPPORTED and can lead to missed transactions.",
+			tx.Hash(), err), "gasPrice", tx.GasPrice, "gasTipCap", tx.GasTipCap, "gasFeeCap", tx.GasFeeCap)
+
+		// Assume success and hand off to the next cycle.
+		return txmtypes.Successful, err
+	}
+	if sendError == nil {
+		return txmtypes.Successful, err
+	}
+	if sendError.IsTransactionAlreadyInMempool() {
+		client.logger.Debugw("Transaction already in mempool", "txHash", tx.Hash, "nodeErr", sendError.Error())
+		return txmtypes.Successful, err
+	}
+	if sendError.IsTemporarilyUnderpriced() {
+		// Broadcaster: If we can't even get the transaction into the mempool at all, assume
+		// success (even though the transaction will never confirm) and hand
+		// off to the ethConfirmer to bump gas periodically until we _can_ get
+		// it in
+		//
+		// Confirmer: Most likely scenario here is a parity node that is rejecting
+		// low-priced transactions due to mempool pressure.
+		// In that case, the safest thing to do is to pretend the transaction
+		// was accepted and continue the normal gas bumping cycle until we can
+		// get it into the mempool
+		client.logger.Infow("Transaction temporarily underpriced", "err", sendError.Error())
+		return txmtypes.Successful, err
+	}
+	if sendError.IsTerminallyUnderpriced() {
+		return txmtypes.Underpriced, err
+	}
+	if sendError.L2FeeTooLow() || sendError.IsL2FeeTooHigh() || sendError.IsL2Full() {
+		return txmtypes.Unsupported, errors.Wrap(sendError, "this error type only handled for L2s")
+	}
+	if sendError.IsNonceTooHighError() {
```

Comment:
Shouldn't sendError.IsL2Full() too belong to this category?
I recall this error can go away automatically